pkg/bb/bbmain/cmd/main.go

```diff
@@ -27,7 +28,96 @@ func main() {
 	run()
 }
 
+// u-root was originally built around the use of symlinks, but not all systems
+// have symlinks. This only recently became an issue with the Plan 9 port.
+//
+// One way to get around this lack, inefficiently, is to make each of the symlinks
+// a small shell script, e.g., on Plan 9, one might have, in /bbin/ls,
+// #!/bin/rc
+// bb ls
+// This leaves a lot to be desired: it puts the execution of a shell in front
+// of each u-root command, and it requires the existence of that shell on the
+// system.
+//
+// The goal is that a single u-root file lead to running the u-root busybox
+// with no intermediate programs running.
+//
+// It is worth taking a look at what a symlink is, how it works in operation,
+// and how we might achieve the same goal some other way.
+//
+// A symlink is plain file, containing 0 or more bytes of text (or utf-8, depending)
+// with an attribute that causes the kernel to give it special treatment.
+// It is not available on all file systems.
+//
+// [Note: they were invented in 1965 for Multics].
+// The symlink is itself still controversial, though widely used.
+//
+// Consider the process of traversing a symlink: it involves the equivalent
+// of stat, open, read, evaluate contents, use that as a file name, repeat as needed.
+//
+// It is possible to get that same effect, with the same overheads, by using #!
+// files but specifying bb as the interpreter.
+//
+// ls would then be:
+// #!/bin/bb ls
+//
+// Note that the absolute path is required, else Linux will throw an error as bb
+// is not in the list of allowed interpreters.
+// The /bin/bb path is not an issue on Plan 9, since users construct their name space
+// on startup and binding /bbin into /bin is no problem.
+//
+// In this case the kernel will stat, open, and read the file, find the executable name,
+// and start it. This approach has as low overhead as the symlink approach.
+//
+// One problem remains: Unix and Plan 9 evaluate arguments in a #! file differently,
+// and, further, invoke the argument in a different way.
+// Given the file shown above, bb on Linux gets the arguments:
```

Comment:
```suggestion
// Given the file shown above, bb on Plan 9 gets the arguments:
```
Is the first example plan 9 and the second linux?

Reply:
yep. Thanks! Fixed.